sdk/src/main/java/me/digi/sdk/DMEPullClient.kt

```diff
@@ -65,12 +65,19 @@ class DMEPullClient(val context: Context, val configuration: DMEPullConfiguratio
                     Pair(true, true),
                     Pair(true, false) -> nativeConsentManager.beginAuthorization(fromActivity, completion)
                     Pair(false, true) -> {
-
                         val consentModeDialogue = ConsentModeSelectionDialogue()
-
-                        consentModeDialogue.show(fromActivity.fragmentManager, "ConsentModeDialogue")
-//                        guestConsentManager.beginGuestAuthorization(fromActivity, completion)
-
+                        consentModeDialogue.configureHandler(object: ConsentModeSelectionDialogue.DecisionHandler {
+                            override fun installDigiMe() {
+                                DMEAppCommunicator.getSharedInstance().requestInstallOfDMEApp(fromActivity) {
+                                    nativeConsentManager.beginAuthorization(fromActivity, completion)
+                                }
+                            }
+
+                            override fun shareAsGuest() {
+                                guestConsentManager.beginGuestAuthorization(fromActivity, completion)
+                            }
+                        })
+                        consentModeDialogue.show(fromActivity.fragmentManager, "ConsentModeSelection")
```

Comment:
Could you call directly _ConsentModeSelectionDialogue()_ and chain _configureHandler_ and _show_ along instead of using _consentModeDialogue_ twice? Maybe it's being used somewhere tho, dunno

Reply:
I could adopt a builder pattern here, would be more code in the `ConsentModeSelectionDialogue` class but cleaner here in the implementation. I like that pattern so will make that change, thanks 👍

Reply:
On closer inspection, a builder pattern isn't so appropriate here due to over complicating the implementation as only one config change is made. Chaining, in my opinion, introduces ambiguity as to what object is being mutated by a given method as without using the builder pattern which is well known and clear as to what's changing, it is anyone's guess.